ui/src/components/vault/VaultManagement/VaultManagement.jsx

```diff
@@ -52,38 +53,38 @@ const VaultManagement = () => {
 
   const { state, walletP } = useApplicationContext();
 
-  const {
-    purses,
-    vaults,
-    vaultToManageId,
-    brandToInfo,
-    treasury: { priceAuthority },
-  } = state;
+  const { purses, vaults, vaultToManageId, brandToInfo, treasury } = state;
 
   /** @type { VaultData } */
-  let vaultToManage = {
-    debt: null,
-    interestRate: null,
-    liquidationRatio: null,
-    locked: null,
-  };
+  let vaultToManage;
   if (vaultToManageId && vaults) {
     vaultToManage = vaults[vaultToManageId];
+  } else {
+    return <Redirect to="/vaults" />;
   }
+  const {
+    interestRate,
+    liquidationRatio,
+    locked,
+    debtSnapshot,
+  } = vaultToManage;
 
-  const { debt, interestRate, liquidationRatio, locked } = vaultToManage;
-
-  if (locked === null || debt === null) {
+  if (!locked || !debtSnapshot) {
```

Comment:
now that these values come from a full `VaultData`, I bet this condition never fails and could be removed.

Reply:
You're right. The only way these could be undefined is if we have a pending/closed/errored vault, which we don't let the user navigate to this "management" view for.